manager/scheduler/scheduler.go

```diff
@@ -286,17 +287,63 @@ func (s *Scheduler) deleteTask(ctx context.Context, t *api.Task) {
 	}
 }
 
+func reconcileGenericResources(nodeRes *api.Resources, assigned map[string]*api.GenericResource) {
+	for taskRes, taskVal := range assigned {
+		nodeVal := nodeRes.Generic[taskRes]
+		reconcileGenericResource(nodeVal, taskVal)
+	}
+}
+
+func reconcileGenericResource(nodeVal, taskVal *api.GenericResource) {
+
+	switch tr := taskVal.Resource.(type) {
+	case *api.GenericResource_Discrete:
+		nr, ok := nodeVal.Resource.(*api.GenericResource_Discrete)
+		if !ok {
+			return
+		}
+
+		if nr.Discrete < tr.Discrete {
+			return
+		}
```

Comment:
I know I said earlier that this should check if `Discrete` would go negative. Now I'm wondering if it's actually better to let it go negative.

The scenario I'm thinking of is if a node is updated to remove some of the available resources, even though it already had tasks using that resource. If `reconcileGenericResource` ignores the resources in use to avoid ending up with a negative value, then when `constraintenforcer` removes the tasks, the `removeTask` in the scheduler will still add back the resources that they had been consuming, and the available value will end up artificially high.